src/ophyd_async/epics/_backend/_aioca.py

```diff
@@ -35,7 +35,58 @@
     dbr.DBR_CHAR: "string",
     dbr.DBR_LONG: "integer",
     dbr.DBR_DOUBLE: "number",
+    dbr.DBR_ENUM: "string",
 }
+_num_dbr = {dbr.DBR_SHORT, dbr.DBR_CHAR, dbr.DBR_LONG, dbr.DBR_FLOAT, dbr.DBR_DOUBLE}
+_floating_dbr = {dbr.DBR_FLOAT, dbr.DBR_DOUBLE}
+
+
+def _data_key_from_augmented_value(source: str, value: AugmentedValue) -> DataKey:
+    """Use the return value of get with FORMAT_CTRL to construct a DataKey
+    describing the signal. See docstring of AugmentedValue for expected
+    value fields by DBR type.
+
+    Args:
+        value (AugmentedValue): Description of the the return type of a DB record
+
+    Returns:
+        DataKey: A rich DataKey describing the DB record
+    """
+
+    # Always present keys
+    if not value.ok:
+        raise Exception(f"{source} error: {value.errorcode}: {str(value)}")
+
+    scalar = value.element_count == 1
+
+    d = DataKey(
+        source=source,
```

Comment:
source that is passed in is value.name with the protocol. Can we get the protocol from elsewhere to not need to pass the source in?

Reply:
protocol is `ca://` for the whole of this file, so fine to put that as a global